lib/circuitry/provisioner.rb

```diff
@@ -0,0 +1,59 @@
+require 'circuitry/queue_creator'
+require 'circuitry/topic_creator'
+require 'circuitry/subscription_creator'
+
+module Circuitry
+  class Provisioner
+    attr_reader :log
+    attr_reader :config
+
+    def initialize(config, logger = Logger.new(STDOUT))
+      @config = config
+      @log = logger
+    end
+
+    def run
+      queue = create_queue
+      topics = create_topics if queue
+      subscribe_topics(queue, topics) if queue && topics
+    end
+
+    private
+
+    def create_queue
+      safe_aws do
+        queue = Circuitry::QueueCreator.find_or_create(
+          config.subscriber_queue_name,
+          dead_letter_queue_name: config.subscriber_dead_letter_queue_name
+        )
+        log.info "Created queue #{queue.url}"
+        queue
+      end
+    end
+
+    def create_topics
+      safe_aws do
+        config.publisher_topic_names.map do |topic_name|
+          topic = Circuitry::TopicCreator.find_or_create(topic_name)
+          log.info "Created topic #{topic.name}"
+          topic
+        end
+      end
+    end
+
+    def subscribe_topics(queue, topics)
+      safe_aws do
+        Circuitry::SubscriptionCreator.subscribe_all(queue, topics)
+        log.info "Subscribed all topics to #{queue.name}"
+        true
+      end
+    end
+
+    def safe_aws
```

Comment:
To make the log more clear about what field, maybe this should take an `action` argument that's just a string prefix for the log message.  e.g.:

```ruby
safe_aws('Create Topics') do
  # ...
end
```

Which would output:

> Create Topics: Access denied. Check your configured credentials.